pkg/api/v20230701preview/openshiftcluster.go

```diff
@@ -133,6 +133,44 @@ const (
 	OutboundTypeLoadbalancer       OutboundType = "Loadbalancer"
 )
 
+// LoadbalancerProfile represents the profile of the cluster public loadbalancer.
+type LoadbalancerProfile struct {
+	// The desired managed outbound IPs for the cluster public load balancer.
+	ManagedOutboundIPs *ManagedOutboundIPs `json:"managedOutboundIps,omitempty" mutable:"true"`
```

Comment:
Calling out my use of a pointer here.  Customer should only be using one of ManagedOutboundIPs, OutboundIPs, or OutboundIPPrefixes.  I could not find a good way to validate if only 1 of these 3 options was within the body of the API request, and validate that ManagedOutboundIPs.Count is between 1-20.  It doesn't seem possible to validate if ManagedOutboundIPs.Count=0 in the body of the API request, or if golang initialized it to its zero value.  I'm open to suggestions if there is a better way to handle this.

Reply:
This is actually a valid way to solve the problem you have described. It is also widely used inside of the Kubernetes (e.g. https://github.com/kubernetes/kubernetes/blob/16534deedf1e3f7301b20041fafe15ff7f178904/staging/src/k8s.io/apiserver/pkg/apis/example/v1/types.go#L84) The only way to be sure that object was not parsed, is to test it's nil.

I am +1 on this

Reply:
same here, if we allow no value and the default value and a custom value then I think this is a fine way to go about it. We are at the restriction of JSON, go defaults, and not having a confusing API to make the distinction in another manner.

Reply:
I appreciate the confirmation that this is OK to do.  I was not sure considering we currently don't use a lot of pointers in the API and I was not sure if there was a way to implement this similar to how maps work in golang (ok is true if key/value is found)